identity/keystone.go

```diff
@@ -63,14 +64,14 @@ func keystoneAuth(client *goosehttp.Client, auth_data interface{}, URL string) (
 	requestData := goosehttp.RequestData{ReqValue: auth_data, RespValue: &accessWrapper}
 	err := client.JsonRequest("POST", URL, "", &requestData, nil)
 	if err != nil {
-		return nil, err
+		return nil, gooseerrors.Newf(err, "", "requesting token failed")
 	}
 
 	details := &AuthDetails{}
 	access := accessWrapper.Access
 	respToken := access.Token
 	if respToken.Id == "" {
-		return nil, fmt.Errorf("authentication failed")
+		return nil, fmt.Errorf("response token Id is empty")
```

Comment:
What happened to the AuthenticationFailed error? This is precisely where we need this error to be returned.
There should be a failing test with this code.